app/store/types.ts

```diff
@@ -0,0 +1,36 @@
+import { store } from './index';
```

Comment:
Im less of a fan of dedicated `types.ts` files. I find that organizing by domain concept and having the types inline make it a bit easier to reason about the code.  What do you think about removing this file and inlining the types where they are used?

Reply:
See comment about the export modules. Having a central source becomes more necessary when we're explicitly defining exports & avoiding circular dependencies